core/src/main/java/org/apache/calcite/sql/SqlCollation.java

```diff
@@ -113,6 +115,31 @@ public SqlCollation(
     this.collationName = generateCollationName(charset);
   }
 
+  /** Encode all the information require to reconstruct a SqlCollection in a SqlList object. */
+  public SqlNodeList asList() {
```

Comment:
Why we can not just use string generated by`SqlCollation.generateCollationName` ? and construct SqlCollation using 
```
 @JsonCreator
  public SqlCollation(
      @JsonProperty("collationName") String collation,
      @JsonProperty("coercibility") Coercibility coercibility) {
```